test/f90_correct/src/apower.f90

```diff
@@ -0,0 +1,58 @@
+!
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+!
+! This test case is test for power operator (int precision).
+
+program test
+  integer :: i1,i2,res(24)
+  integer :: eres(24) = [0, 0, 0, 0, 1, 1, 1, 1, 256, 256, 256, &
+                         256, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
+  i2 = -8
+
+  i1 = 2
+  res(1) = i1 ** i2
+  res(2) = i1 ** (-8)
+  i1 = -2
+  res(3) = i1 ** i2
+  res(4) = i1 ** (-8)
+  i1 = -1
+  res(5) = i1 ** i2
+  res(6) = i1 ** (-8)
```

Comment:
When I try with optimisations, only these two fail. Everything else matches. 
Should there be better tests?

Reply:
The testcase has been modified to be more readable and clear.